run-tests.pl

```diff
@@ -359,6 +359,20 @@ (&)
    }  until => sub { $_[0]->get };
 }
 
+# wrapper around Future::with_cancel which works around an apparent bug:
+#
+# the returned future does not keep a reference to the old future, which means
+# it may get garbage-collected.
```

Comment:
Hm. That sounds like a potential Future bug.

Reported https://rt.cpan.org/Ticket/Display.html?id=122920

Reply:
thanks. added to comment. 7997669